pkg/apis/externaldns/validation/validation.go

```diff
@@ -69,5 +69,15 @@ func ValidateConfig(cfg *externaldns.Config) error {
 	if cfg.IgnoreHostnameAnnotation && cfg.FQDNTemplate == "" {
 		return errors.New("FQDN Template must be set if ignoring annotations")
 	}
+
+	if cfg.ServicePublishIPsType != "" {
+		switch cfg.ServicePublishIPsType {
+		case "public", "private":
```

Comment:
I don't understand why we have a switch case here if we then only handle the default case. Can you change it with just a simpler to read if statement?